client/src/common/components/Activities/components/Activity.jsx

```diff
@@ -1,79 +1,137 @@
-        </Fragment>
-      );
-    }
+  renderListLink = () => {
+    const {
+      activity: {
+        list: { id, name }
+      }
+    } = this.props;
+
+    return (
+      <Link className="activity__link" to={`/sack/${id}`}>
+        {name}
+      </Link>
+    );
   };
 
-  renderListInfo = () => {
+  renderItemActivity = () => {
     const {
-      activity: { list }
+      activity: { activityType, cohort, editedValue, item, list, performer }
     } = this.props;
 
-    if (list) {
-      const { listId, listName } = list;
+    return (
+      <Fragment>
+        <FormattedMessage
+          id={activityType}
+          values={{
+            item: item ? <em>{item.name}</em> : null,
+            performer: <em>{performer.name}</em>,
+            value: editedValue
+          }}
+        />
+        {list && (
+          <FormattedMessage
+            id="activity.list"
+            values={{ list: this.renderListLink() }}
+          />
+        )}
+        {cohort && (
+          <FormattedMessage
+            id="activity.cohort"
+            values={{ cohort: this.renderCohortLink() }}
+          />
+        )}
+      </Fragment>
+    );
+  };
 
-      return (
-        <Fragment>
-          <Link className="activity__link" to={`/sack/${listId}`}>
-            {listName}
-          </Link>
-        </Fragment>
-      );
-    }
+  renderListActivity = () => {
+    const {
+      activity: { activityType, cohort, editedUser, editedValue, performer }
+    } = this.props;
+
+    return (
+      <Fragment>
+        <FormattedMessage
+          id={activityType}
+          values={{
+            list: <em>{this.renderListLink()}</em>,
+            performer: <em>{performer.name}</em>,
+            user: editedUser ? editedUser.name : null,
+            value: editedValue
+          }}
+        />
+        {cohort && (
+          <FormattedMessage
+            id="activity.cohort"
+            values={{ cohort: this.renderCohortLink() }}
+          />
+        )}
+      </Fragment>
+    );
+  };
+
+  renderCohortActivity = () => {
+    const {
+      activity: { activityType, editedUser, editedValue, performer }
+    } = this.props;
+
+    return (
+      <FormattedMessage
+        id={activityType}
+        values={{
+          cohort: <em>{this.renderCohortLink()}</em>,
+          performer: <em>{performer.name}</em>,
+          user: editedUser ? editedUser.name : null,
+          value: editedValue
+        }}
+      />
+    );
   };
 
   render() {
     const {
-      activity: {
-        performer: { performerAvatarUrl, performerName },
-        createdAt,
-        item: { itemName }
-      }
+      activity: { cohort, createdAt, item, list, performer }
     } = this.props;
-    const date = dateFromString(createdAt);
 
     return (
       <div className="activity">
         <div className="activity__avatar">
           <Avatar
-            avatarUrl={performerAvatarUrl}
+            avatarUrl={performer ? performer.avatarUrl : null}
```

Comment:
hmm, if the user has no avatar or there is something wrong with avatarUrl an avatar placeholder is displayed, so  activity layout is the same because the user can't remove an account so I removed this check